waspc/src/Analyzer/TypeChecker/TypeError.hs

```diff
@@ -0,0 +1,37 @@
+module Analyzer.TypeChecker.TypeError
```

Comment:
I don't know much about how errors are reported during type checking / in compilers -> how do you feel about this solution, do you think it is good, just a start, ... ?

I mostly wonder if errors are detailed enough to enable us to report a meaningful error message to the user. For example, right now we don't have any information on the token where this error might be reported, right? That is something we would need, correct? Is there any other information potentially missing? I guess we also need some kind of mechanism that will produce meaningful messages based on these errors -> stuff like "weaken" and "unification" are implementation details for users, they shouldn't have to know what they mean.

I am pretty sure you are aware of all of this and you left it for later, but I think it would be good to write one TODO encompassing this stuff and documentating current state, what is missing, some ideas on what we want it to look like in the future, and so on. That way we covered that well and we can feel ok knowing that there is a plan that somebody else can execute on if not us.